src/main/java/uk/gov/hmcts/ccd/auditlog/AuditEntry.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    public void setIdamId(String idamId) {
+        this.idamId = idamId;
+    }
+
+    public String getInvokingService() {
+        return invokingService;
+    }
+
+    public void setInvokingService(String invokingService) {
+        this.invokingService = invokingService;
+    }
+
+    public static String getTAG() {
+        return TAG;
+    }
+
+    public String getCaseId() {
+        return caseId;
+    }
+
+    public void setCaseId(String caseId) {
+        this.caseId = caseId;
+    }
+
+    public String getJurisdiction() {
+        return jurisdiction;
+    }
+
+    public void setJurisdiction(String jurisdiction) {
+        this.jurisdiction = jurisdiction;
+    }
+
+    public String getCaseType() {
+        return caseType;
+    }
+
+    public void setCaseType(String caseType) {
+        this.caseType = caseType;
+    }
+
+    public String getEventSelected() {
+        return eventSelected;
+    }
+
+    public void setEventSelected(String eventSelected) {
+        this.eventSelected = eventSelected;
+    }
+
+    public String getRequestId() {
+        return requestId;
+    }
+
+    public void setRequestId(String requestId) {
+        this.requestId = requestId;
+    }
+
+    public String getTargetIdamId() {
+        return targetIdamId;
+    }
+
+    public void setTargetIdamId(String targetIdamId) {
+        this.targetIdamId = targetIdamId;
+    }
+
+    public List<String> getTargetCaseRoles() {
+        return targetCaseRoles;
+    }
+
+    public void setTargetCaseRoles(List<String> targetCaseRoles) {
+        this.targetCaseRoles = targetCaseRoles;
+    }
+
+    @Override
+    public String toString() {
+        return TAG + " " + dateTime + " operationType:" + operationType +
+            (isNotBlank(caseId) ? ", caseId:" + caseId : "") +
+            (isNotBlank(idamId) ? ", idamId:" + idamId : "") +
+            (isNotBlank(invokingService) ? ", invokingService:" + invokingService : "") +
+            (isNotBlank(path) ? ", endpointCalled:" + httpMethod + " " + path : "") +
+            ", operationOutcome:" + httpStatus +
+            (isNotBlank(caseType) ? ", caseType:" + caseType : "") +
+            (isNotBlank(jurisdiction) ? ", jurisdiction:" + jurisdiction : "") +
+            (isNotBlank(eventSelected) ? ", eventSelected:" + eventSelected : "") +
+           (isNotBlank(targetIdamId) ? ", idamIdOfTarget:" + targetIdamId : "") +
+//            (isNotBlank(listOfCaseTypes) ? ", listOfCaseTypes:" + listOfCaseTypes : "") +
+            (targetCaseRoles != null && !targetCaseRoles.isEmpty() ? ", targetCaseRoles:" +
+                targetCaseRoles.stream().map(String::toString).collect(Collectors.joining(",")) : "") +
+            (isNotBlank(requestId) ? ", X-Request-ID:" + requestId : "") +
+            '}';
+    }
+
+    public String formattedAuditData() {
+        return  ReflectionToStringBuilder.toString(this, JSON_STYLE, false, false,
+            true, null).toString();
+    }
```

Comment:
I would move this logic into a separate AuditEntryFormatter class for better separation of concerns. We could also more easily change the formatting

Reply:
done